### DISCUSSION TITLE
Frontend step-by-step

### DISCUSSION THREAD

**Discussion by giangiac** [tqec/tqec#87] [General]
Hi TQEC friends,

I am unfamiliar with JavaScript, React, Pixi, ... but would like to help developing the frontend part.
I opted for a hands-on approach to learn those language & libraries, with the side benefit of familiarizing with the frontend codebase.

I started by setting up a template React project with:
https://create-react-app.dev/
and then continued by adding snippets of code from the tqecjs frontend:
https://github.com/giangiac/tqec/tree/test-frontend/step-by-step-frontend

I noticed that this may be helpful to the community members who, like me, need to learn the basics of a web app.
If there is interest, I can create a PR to track the extension of the "step-by-step frontend coding", with the commit message indicating the functionality added by each of them.
Checking the commit content will help locating the part of code responsible for that functionality.

**Comment by smburdick** [comment]
This looks like a really helpful way to ramp up! We can discuss it in today's meeting

**Reply by giangiac** [reply]
As for today's call, I opened a PR (as draft): https://github.com/QCHackers/tqec/pull/89
